pkg/migrations/sql/idempotent/exemplar.sql

```diff
@@ -0,0 +1,122 @@
+-- get_exemplar_label_positions returns the position of label_keys as a one-to-one mapping with label_keys. It returns
+-- the positions of all label keys corresponding to that metric, so that it remains easier to add null values to those indexes
+-- whose labels are not present in the exemplar being inserted at the golang level.
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_or_create_exemplar_label_key_positions(metric_name_text TEXT, label_keys TEXT[])
+RETURNS TABLE (metric_family_text TEXT, label_positions_map JSON) AS
+$$
+DECLARE
+    current_position INTEGER := 1; -- index in postgres starts from 1. Let's maintain the convention for less confusion.
+    new_position INTEGER;
+    k TEXT;
+BEGIN
+    LOCK TABLE SCHEMA_CATALOG.exemplar_label_key_position IN ACCESS EXCLUSIVE MODE;
+    -- If there isn't any data for the given metric_name_text.
+    IF ( SELECT count(key) = 0 FROM SCHEMA_CATALOG.exemplar_label_key_position WHERE metric_name=metric_name_text ) THEN
```

Comment:
Since `key` cannot be `NULL`, should just do `count(*)`. Postgres would probably optimize this away but its clearer to put `count(*)` for counting rows if you don't care if a certain column is `NULL` or not.

Reply:
I though using `*` makes postgres not use the indices. Hence, I used a column that had the index applied only, making it an index-only scan.

Reply:
That's a micro-optimization at best, I'd stick with the standard `count(*)` since its commonly used across the SQL codebase.